corpora/csip/metadata/mets-root/CSIP4/testCase.xml

```diff
@@ -1,23 +1,161 @@
+  </description>
   <dependencies>
+    <dependency requirementId="VocabularyContentInformationTypeSpecification" URL="https://earkcsip.dilcis.eu/schema/CSIPVocabularyContentInformationType.xml">
+      The value of the attribute MUST be from the vocabulary.
+    </dependency>
   </dependencies>
-  <!-- A list of the validation rules derived from the test case -->
   <rules>
+    <rule id="1">
+      <description>
+        If the METS document is in the root folder, mets/@csip:CONTENTINFORMATIONTYPE SHOULD exist.
+      </description>
+      <error level="WARNING">
+        <message>
+          mets/@csip:CONTENTINFORMATIONTYPE attribute does not exist. SHOULD be: mets/@csip:CONTENTINFORMATIONTYPE exists and has a value from VocabularyContentInformationTypeSpecification.
+        </message>
+      </error>
+      <corpusPackages>
+        <package name="root_mets_file_mets-xml_mets_csip-CONTENTINFORMATIONTYPE_not_exist" isValid="FALSE">
+          <path></path>
+          <description>
+            METS.xml file is located in the root folder.
+            The METS.xml file attribute mets/@csip:CONTENTINFORMATIONTYPE does not exist.
+          </description>
+        </package>
+        <!--  It seems most efficient to have one valid package for all SHOULD 
+              requirements - there are about 20 of them, so the alternative would
+              be to have 20 rule-specific valid packages. -->
+        <package name="minimal_IP_with_1_rep_plus_mets-xml_SHOULD_nodes" isValid="TRUE">
+          <path></path>
+          <description>
+            Minimal IP that matches all MUST requirements, plus all SHOULD requirements (both for elements and attributes).
+          </description>
+        </package>
+      </corpusPackages>
+    </rule>
+    <rule id="2">
+      <description>
+        If the METS document is in the representations folder, mets/@csip:CONTENTINFORMATIONTYPE MUST exist.
+      </description>
+      <error level="ERROR">
+        <message>
+          mets/@csip:CONTENTINFORMATIONTYPE attribute does not exist. MUST be: mets/@csip:CONTENTINFORMATIONTYPE exists and has a value from VocabularyContentInformationTypeSpecification.
+        </message>
+      </error>
+      <corpusPackages>
+        <package name="rep_mets_file_mets-xml_mets_csip-CONTENTINFORMATIONTYPE_not_exist" isValid="FALSE">
+          <path></path>
+          <description>
+            METS.xml file is located in the /representations folder.
+            The METS.xml file attribute mets/@csip:CONTENTINFORMATIONTYPE does not exist.
```

Comment:
I think that the package needs to have missing attributes at both root and representation level to make sure that we see the expected different behavior on both levels... (warning at root level and error at representation level).

Reply:
Following your suggestion, I started to create the description of such a package, but then realised there is no good place for that because it violates two different rules at the same time. That's because my current approach is to have rule 1 for root METS file and rule 2 for representation METS. There might be value in having a package for the two combined, but that would go against our principle of violating only one rule at a time and open the door to a large number of other cases where violating several rules might seem reasonable. I think that would divert our attention away from vital stuff to "nice-to-have," so I'd avoid it 🙂.

Reply:
I would say it still just violates 1 rule and does it twice - but okay ;-)